packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -90,9 +90,10 @@ function convertBlockScopedToVar(path, parent, scope, moveBindingsToParent = fal
     const parentScope = scope.getFunctionParent();
     const ids = path.getBindingIdentifiers();
     for (let name in ids) {
-      scope.removeOwnBinding(name);
+      let binding = scope.getOwnBinding(name);
+      if (binding) binding.kind = "var";
+      scope.moveBindingTo(name, parentScope);
```

Comment:
there is a similar if statement inside `moveBindingTo`, either way should work. Let me know if you have a preference. https://github.com/babel/babel/blob/3d5969ecff25bc8c581ad3ec3cfff1cba41fd3b0/packages/babel-traverse/src/scope/index.js#L884